### PR TITLE
Prefer system tar, instead of tar in path

### PR DESCRIPTION
**Pull request by sigurdm**
This should fix: https://github.com/dart-lang/pub/issues/2286

### CODE REVIEW

File: lib/src/io.dart [hunk 1]
```diff
@@ -887,7 +906,7 @@ Future extractTarGz(Stream<List<int>> stream, String destination) async {
 /// silenced by this flag.
 final bool _noUnknownKeyword = _computeNoUnknownKeyword();
 bool _computeNoUnknownKeyword() {
-  if (!Platform.isLinux) return false;
+  if (Platform.isLinux) return false;
```
Comment (jonasfj) [inline lib/src/io.dart]: I don't think this is intended

File: lib/src/io.dart [hunk 2]
```diff
@@ -842,10 +842,28 @@ Future<HttpServer> bindServer(String host, int port) async {
...
+      'Could not find a system `tar` installed in /bin/tar or /usr/bin/tar, '
+      'attempting to use tar from PATH');
+  return 'tar';
+}
+
 /// Extracts a `.tar.gz` file from [stream] to [destination].
 Future extractTarGz(Stream<List<int>> stream, String destination) async {
   log.fine('Extracting .tar.gz stream to $destination.');
-  if (Platform.isWindows) {
+  if (!Platform.isWindows) {
```
Comment (jonasfj) [inline lib/src/io.dart]: I don't think this change is intentional..

File: lib/src/io.dart [hunk 3]
```diff
@@ -842,10 +842,28 @@ Future<HttpServer> bindServer(String host, int port) async {
...
 
+/// Find a tar. Prefering system installed tar.
+///
+/// On linux tar should always be /bin/tar [See FHS 2.3][1]
+/// On MacOS it seems to always be /usr/bin/tar.
+///
+/// [1]: https://refspecs.linuxfoundation.org/FHS_2.3/fhs-2.3.pdf
+String _findTarPath() {
+  for (final file in ['/bin/tar', '/usr/bin/tar']) {
+    if (File(file).existsSync()) {
```
Comment (jonasfj) [inline lib/src/io.dart]: nit: use `fileExists(file)` defined in this file and use throughout pub.

File: lib/src/io.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/io.dart]: indeed :)

File: lib/src/io.dart (hunk 3 above)
Comment (sigurdm) [reply lib/src/io.dart]: done

**Comment by sigurdmeldgaard** [comment]
The value should be cached...

**Comment by sigurdm** [comment]
> The value should be cached...

Done

**Pull request merged**